src/components/radios/_radios.scss

```diff
@@ -186,51 +194,128 @@
-  }
+  .govuk-radios--small {
 
-  .govuk-radios--small .govuk-radios__input + .govuk-radios__label::after {
-    top: 14px;
-    left: 6px;
-    width: 9px;
-    height: 3.5px;
-  }
+    $input-offset: ($govuk-touch-target-size - $govuk-small-radios-size) / 2;
+    $label-offset: $govuk-touch-target-size - $input-offset;
 
-  .govuk-radios--small .govuk-radios__label {
-    margin-top: -2px;
-    padding: 13px govuk-spacing(3) 13px 0;
-    float: left;
+    .govuk-radios__item {
+      @include govuk-clearfix;
+      min-height: 0;
+      margin-bottom: 0;
+      padding-left: $label-offset;
+      float: left;
+    }
 
-    @include govuk-media-query($from: tablet) {
-      padding: 11px govuk-spacing(3) 10px 0;
+    // Shift the touch target into the left margin so that the visible edge of
+    // the control is aligned
+    //
+    //   ┆Which colour is your favourite?
+    //  ┌┆───┐
+    //  │┆() │ Purple
+    //  └┆▲──┘
+    //  ▲┆└─ Radio pseudo element, aligned with margin
+    //  └─── Touch target (invisible input), shifted into the margin
+    .govuk-radios__input {
+      left: $input-offset * -1;
     }
 
-  }
+    // Adjust the size and position of the label.
+    //
+    // Unlike larger radios, we also have to float the label in order to
+    // 'shrink' it, preventing the hover state from kicking in across the full
+    // width of the parent element.
+    .govuk-radios__label {
+      margin-top: -2px;
+      padding: 13px govuk-spacing(3) 13px 1px;
+      float: left;
+
+      @include govuk-media-query($from: tablet) {
+        padding: 11px govuk-spacing(3) 10px 1px;
+      }
+    }
 
-  .govuk-radios--small .govuk-radios__item {
-    min-height: 0;
-    padding-left: 35px;
-  }
+    // ( ) Radio ring
+    //
+    // Reduce the size of the control [1], vertically centering it within the
+    // touch target [2]
+    .govuk-radios__label::before {
+      top: $input-offset - $govuk-border-width-form-element; // 2
+      width: $govuk-small-radios-size; // 1
+      height: $govuk-small-radios-size; // 1
+    }
 
-  .govuk-radios--small .govuk-radios__input + .govuk-radios__label::after {
-    width: 10px;
-    height: 10px;
-    border-width: 1px;
-  }
+    //  •  Radio button
+    //
+    // Reduce the size of the 'button' and center it within the ring
+    .govuk-radios__label::after {
+      top: 14px;
+      left: 6px;
+      width: 10px;
+      height: 10px;
+      border-width: 1px;
+    }
 
-  .govuk-radios--small .govuk-radios__item:hover > .govuk-radios__label::before {
-    box-shadow: 0 0 0 10px govuk-colour("grey-3");
-  }
+    // Fix position of hint with small radios
+    //
+    // Do not use hints with small radios – because they're within the input
+    // wrapper they trigger the hover state, but clicking them doesn't actually
+    // activate the control.
+    //
+    // (If you do use them, they won't look completely broken... but seriously,
+    // don't use them)
+    .govuk-radios__hint {
```

Comment:
I wonder if 'pointer-events: none' would ensure it doesnt trigger the hover state...

Reply:
I'd had a similar thought, but sadly not, as the pointer events just 'go through' the hint and find the label.